libsolidity/ast/Types.cpp

```diff
@@ -740,6 +740,18 @@ TypeResult IntegerType::binaryOperatorResult(Token _operator, Type const* _other
 		return this;
 	}
 
+	if (auto const* other = dynamic_cast<RationalNumberType const*>(_other); other)
+	{
+		if (_operator == langutil::Token::Mod || _operator == langutil::Token::Div)
+		{
+			std::optional<rational> value = ConstantEvaluator::evaluateBinaryOperator(_operator, {0}, other->value());
+			if (!value)
+			{
+				return nullptr;
+			}
```

Comment:
Maybe we could already do:
```suggestion
			if (std::optional<rational> value = ConstantEvaluator::evaluateBinaryOperator(_operator, {0}, other->value()); !value)
				return nullptr;
```
And then you would remove the curly brackets from all ifs, i.e.:
```c++
	if (auto const* other = dynamic_cast<RationalNumberType const*>(_other); other)
		if (_operator == langutil::Token::Mod || _operator == langutil::Token::Div)
			if(std::optional<rational> value = ConstantEvaluator::evaluateBinaryOperator(_operator, {0}, other->value()); !value)
				return nullptr;
```

We usually don't use them for if-Statements that have only one statement in their body :)

Reply:
Ah, habits got the better of me :) thanks for the hint!